### ISSUE TITLE
New constraints in LTS Haskell 1.1

### ISSUE THREAD

**Issue by snoyberg** [fpco/lts-haskell#4]
These constraints are necessary because a minor version bump in a packages depends on a major version bump in a dependency.

monad-control-0.3.3.0 depended on by:
- hoauth2-0.4.4 (>=1 && < 1.1). No maintainer. @freizl
- scotty-0.9.1 (>=1.0.0.0 && < 1.1). Andrew Farmer <afarmer@ittc.ku.edu>. @scotty-web

rest-core-0.33.1.2 depended on by:
- rest-happstack-0.2.10.5 (==0.34.*). Silk <code@silk.co>. @bergmark @hesselink
- rest-snap-0.1.17.15 (==0.34.*). Silk <code@silk.co>. @bergmark @hesselink
- rest-wai-0.1.0.5 (==0.34.*). Silk <code@silk.co>. @bergmark @hesselink

**Comment by hesselink** [comment]
@snoyberg What do you want us to do here?

**Comment by snoyberg** [comment]
The ideal would be releasing point releases that are compatible with old versions of rest-core. Without that, LTS Haskell 1.X won't be able to pick up any new versions of the rest-happstack, rest-snap, and rest-wai packages.

**Comment by snoyberg** [comment]
Another possibility would be major version bumping when you require a new major version of a dependency, that way you still have a means of backporting bugfixes to stable versions of your packages.

**Comment by bergmark** [comment]
You can introduce a 5th version component for that

**Comment by snoyberg** [comment]
::shrug:: I'm not going to try to convince you to change the way you're versioning your packages. You can see the commit I had to make on this issue. If you're comfortable with that, you needn't change anything.

**Comment by hesselink** [comment]
tl;dr; I think we should either keep the current situation or remove rest from LTS.

Ah I see, you want to allow future point release of e.g. rest-happstack etc. without allowing the new major version of rest-core. I'm uncomfortable with doing a major bump of these packages, since they are completely compatible. Major bumps cause a lot of work for people using our packages, so I'd rather avoid them. We could allow older versions of rest-core, but that would require more CPP and result in something that probably won't be tested by us. We could try to backport future important changes, but to be honest, we'd probably forget.

Another option would be to remove rest-* from LTS. There's still going to be some backward incompatible changes to these packages. We try to keep the changes minimal (especially since large changes will cause a lot of work for us, since we're the largest user) but sometimes they're not completely compatible, like in this case.

**Comment by bergmark** [comment]
> Ah I see, you want to allow future point release of e.g. rest-happstack etc. without allowing the new major version of rest-core.

In case it wasn't clear this is what i was proposing to solve by using a 5th version component, if we find that we'd like to backport something in e.g. rest-happstack to get it into LTS we can release `rest-happstack-0.2.10.4.1`. But I'll probably not do this unless requested by a user of this starts causing issues for stackage.

**Comment by snoyberg** [comment]
Just to state this openly, though I think it's probably understood: I don't have *any* hard ideas on what must be done for "LTS Haskell" compliance. The point of this first series of releases is to shake out problems, such as this one. Granted, this specific kind of problem is one that I expected, but dealing with it in the real world is much better than discussing it abstractly.

I'm fine with the proposal you guys are giving right now (i.e., leave version numbers as they are and use the 5th component if necessary). It'll be interesting to see if there are requests from users for something else.

Don't take me opening an issue as demanding a change, but simply as a comment that "there is an issue."

On the other hand, I *do* think for hoauth2 and scotty it would make a lot of sense to release versions that are compatible with the old monad-control.
